### ISSUE TITLE
Periodic BCs Don't Transfer to RestrictedDiscreteModel

### ISSUE THREAD

**Issue by Omega-xyZac** [gridap/Gridap.jl#516]
Hi all,
It appears that periodic boundary conditions do not transfer from the `CartesianDiscreteModel` to a `RestrictedDiscreteModel`. It would be great to fix this as it would speed-up homogenisation code for porous materials. Code below:
```
using Gridap, Gridap.FESpaces, Gridap.Fields, Gridap.ReferenceFEs,
   Gridap.Geometry, Gridap.TensorValues, Gridap.CellData, Gridap.Integration,
   Gridap.Algebra, Gridap.Arrays

n = 64
struc = zeros(n,n,n);
C = Int(n/2);
W = Int(round(7*n/32));
struc[C-W:C+W,C-W:C+W,:] .= 1
struc[C-W:C+W,:,C-W:C+W] .= 1
struc[:,C-W:C+W,C-W:C+W] .= 1

nely,nelx,nelz = size(struc,1),size(struc,2),size(struc,3);
# Flatten density array into vector to multiply weak form
density = Float64[];
for i = 1:nelz
    append!(density, collect(Iterators.flatten(struc[:,:,i]')))
end
# Generate model
domain = (0.0, 1, 0.0, 1, 0.0, 1);
model = CartesianDiscreteModel(domain,(nelx,nely,nelz); isperiodic=(true,true,true));
trian = Triangulation(model)
# Mask only dense cells
function is_in_hom(id)
  density[id]>0
end
oldcell_to_ids = get_cell_to_bgcell(trian)
oldcell_to_is_in = collect1d(lazy_map(is_in_hom,oldcell_to_ids))
model_in = DiscreteModel(model,oldcell_to_is_in)

writevtk(model_in,"./model_in")
writevtk(model,"./model")
```
This generates
![PBCsbug](https://user-images.githubusercontent.com/60646897/105232495-323c8500-5bb4-11eb-9ab1-2c80bf83e333.png)
where the white portions are the `CartesianDiscreteModel` with PBCs and the red section is also the face model data for `model_in` (`RestrictedDiscreteModel`). It appears that the PBCs are no longer present here.

**Comment by fverdugo** [comment]
@Omega-xyZac Thanks for reporting!

The problem is that we are generating a new GridTopology for the restricted part, instead of restricting the given GridTopology
https://github.com/gridap/Gridap.jl/blob/b9ed605b089e0836d89abfe2f682357bdce6b24f/src/Geometry/DiscreteModelPortions.jl#L26

The result is that we lose info about periodic BCs.

**Comment by fverdugo** [comment]
![fixed](https://user-images.githubusercontent.com/48508177/105358682-ca5b5e00-5bf6-11eb-83af-9693318d7570.png)

Fixed!

**Comment by fverdugo** [comment]
Fixed in #517

**Comment by Omega-xyZac** [comment]
Wonderful! :D
